openc3/lib/openc3/models/script_status_model.rb

```diff
@@ -0,0 +1,202 @@
+      end_time: nil, # The time the script ended ISO format
+      disconnect: false,
+      environment: nil,
+      suite_runner: false,
+      errors: nil,
+      pid: nil,
+      updated_at: nil,
+      scope:
+    )
+      @state = state
+      if is_complete?()
+        super("#{COMPLETED_PRIMARY_KEY}__#{scope}", name: name, updated_at: updated_at, plugin: nil, scope: scope)
+      else
+        super("#{RUNNING_PRIMARY_KEY}__#{scope}", name: name, updated_at: updated_at, plugin: nil, scope: scope)
+      end
+      @shard = shard.to_i
+      @filename = filename
+      @current_filename = current_filename
+      @line_no = line_no
+      @username = username
+      @user_full_name = user_full_name
+      @start_time = start_time
+      @end_time = end_time
+      @disconnect = disconnect
+      @environment = environment
+      @suite_runner = suite_runner
+      @errors = errors
+      @pid = pid
+    end
+
+    def is_complete?
+      return (@state == 'complete' or @state == 'complete_errors' or @state == 'stopped' or @state == 'crash' or @state == 'killed')
+    end
+
+    def state=(new_state)
+      # If the state is already a flavor of complete, leave it alone (first wins)
+      if not is_complete?()
+        @state = new_state
+        # If setting to complete, check for errors
+        # and set the state to complete_errors if they exist
+        if @state == 'complete' and @errors
+          @state = 'complete_errors'
+        end
+      end
+    end
+
+    # Update the Redis hash at primary_key and set the field "name"
+    # to the JSON generated via calling as_json
+    def create(update: false, force: false, queued: false, isoformat: true)
+      @updated_at = Time.now.utc.to_nsec_from_epoch
+
+      if queued
+        write_store = self.class.store_queued
+      else
+        write_store = self.class.store
+      end
+      write_store.hset(@primary_key, @name, JSON.generate(self.as_json(:allow_nan => true), :allow_nan => true))
+
+      # Also add to ordered set on create
+      write_store.zadd(@primary_key + "__LIST", @name.to_f, @name) if not update
+    end
+
+    def update(force: false, queued: false)
+      # Magically handle the change from running to completed
+      if is_complete?() and @primary_key == "#{RUNNING_PRIMARY_KEY}__#{@scope}"
+        # Destroy the running key
+        destroy()
+        @destroyed = false
+
+        # Move to completed
+        @primary_key = "#{COMPLETED_PRIMARY_KEY}__#{@scope}"
+        create(update: false, force: force, queued: queued, isoformat: true)
+      else
+        create(update: true, force: force, queued: queued, isoformat: true)
+      end
+    end
+
+    # Delete the model from the Store
+    def destroy
+      @destroyed = true
+      undeploy()
+      self.class.store.hdel(@primary_key, @name)
+      # Also remove from ordered set
+      self.class.store.zremrangebyscore(@primary_key + "__LIST", @name.to_f - 0.1, @name.to_f + 0.1)
```

Comment:
`@name` is the PID? Isn't this an integer? Do you have to do the +- 0.1 or does it work with just the `@name.to_i`?

Reply:
Name is the script id not PID.   
Redis store values are floats, so I figured it would safer with the range rather than exact match.